engine/src/lib.rs

```diff
@@ -285,6 +291,32 @@ mod contract {
         io.return_output(&data[..]);
     }
 
+    /// Sets the flag to pause the contract.
+    #[no_mangle]
+    pub extern "C" fn pause_contract() {
+        let mut io = Runtime;
+        let mut state = state::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        if state.is_paused {
```

Comment:
You can use `require_running` instead of `if ...` here.

Reply:
Yeah, but I kind of preferred to do it explicitly here so this code is symmetrical to the resume_contract method.